BraintreeCore/src/main/java/com/braintreepayments/api/core/PaymentMethod.kt

```diff
@@ -1,102 +1,35 @@
 package com.braintreepayments.api.core
 
-import android.os.Parcel
 import androidx.annotation.RestrictTo
 import org.json.JSONException
 import org.json.JSONObject
 
 /**
- * An abstract class to extend when creating a payment method. Contains logic and
- * implementations shared by all payment methods.
+ * [PaymentMethod] represents the common interface of all payment method.
+ *
+ * @property sessionId The session id associated with this request. The session is a uuid.
+ * This field is automatically set at the point of tokenization, and any previous
+ * values ignored.
+ * @property source The source associated with the tokenization call for analytics use. Set automatically.
+ * @property integration The integration method associated with the tokenization call for analytics use.
+ * Defaults to custom and does not need to ever be set.
+ * @property apiPath The Api Path of the card
  */
-abstract class PaymentMethod {
+interface PaymentMethod {
 
-    /**
-     * @suppress
-     */
-    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-    companion object {
+    var sessionId: String?
+    var source: String?
+    var integration: IntegrationType?
+    val apiPath: String
```

Comment:
Should `apiPath` also be a var?

Reply:
The value for apiPath is fixed for the type of PaymentMethod, for example, in VenmoAccount, the implementation is 
    public String getApiPath() {
        return "venmo_accounts";
    }